Several missing LAYOUTS to rules.mk

arcmags opened the pull request:
## Description

Several missing LAYOUTS variables were added to *keyboards/dztech/dz60_rgb\*/rules.mk* files.  These changes are
necessary if one wants to be able to use any corresponding layouts in *layouts/community/*.

### Sources

+ https://kbdfans.com/collections/pcb/products/dz60rgb-ansi-mechanical-keyboard-pcb
+ https://kbdfans.com/collections/pcb/products/dz60rgb-hot-swap-custom-keyboard-pcb

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

drashna commented:
You have a bunch of merge conflicts here.

fauxpark commented:
Merge conflict artifacts.

fauxpark commented on layouts/layouts:
```diff
@@ -0,0 +1 @@
+/home/mags/user/dat/qmk/layouts
```
Please remove this symlink.

arcmags replied:
done

fauxpark commented on keyboards/dztech/dz60rgb/keymaps/didel/keymap.c:
```diff
@@ -26,31 +26,43 @@ enum didel_layers {
 };
 
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+<<<<<<< HEAD
+    [_QWERTY] = LAYOUT_60_ansi_arrow(
+        KC_GESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSPC,
+=======
     [_QWERTY] = LAYOUT(
         QK_GESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSPC,
+>>>>>>> upstream/develop
```
```suggestion
    [_QWERTY] = LAYOUT_60_ansi_arrow(
        QK_GESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSPC,
```

fauxpark commented on keyboards/dztech/dz60rgb/keymaps/matthewrobo/keymap.c:
```diff
@@ -31,57 +31,68 @@ extern bool autoshift_enabled;
 #define LM_LALT LM(_FNM, MOD_LALT)
 
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+<<<<<<< HEAD
+	[_QWERTY] = LAYOUT_60_ansi_arrow(
+		KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, \
+=======
 	[_QWERTY] = LAYOUT(
 		QK_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, \
+>>>>>>> upstream/develop
```
```suggestion
	[_QWERTY] = LAYOUT_60_ansi_arrow(
		QK_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, \
```

fauxpark commented on keyboards/dztech/dz60rgb/keymaps/matthewrobo/keymap.c:
```diff
@@ -31,57 +31,68 @@ extern bool autoshift_enabled;
 		_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, \
 		_______, _______, _V_V_V_,                   _______,                   _______, _______,          _______, _______, _______  \
 	),
+<<<<<<< HEAD
+	[_NAV] = LAYOUT_60_ansi_arrow(
+		KC_NLCK, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_P7,   KC_P8,   KC_P9,   KC_PAST, KC_PMNS, KC_PPLS, KC_DEL,  \
+		_______, KC_PGUP, KC_HOME, KC_UP,   KC_END,  _______, _______, KC_P4,   KC_P5,   KC_P6,   KC_PSLS, KC_PSCR, KC_SLCK, KC_INS,  \
+=======
 	[_NAV] = LAYOUT(
 		KC_NUM,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_P7,   KC_P8,   KC_P9,   KC_PAST, KC_PMNS, KC_PPLS, KC_DEL,  \
 		_______, KC_PGUP, KC_HOME, KC_UP,   KC_END,  _______, _______, KC_P4,   KC_P5,   KC_P6,   KC_PSLS, KC_PSCR, KC_SCRL, KC_INS,  \
+>>>>>>> upstream/develop
```
```suggestion
	[_NAV] = LAYOUT_60_ansi_arrow(
		KC_NUM,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_P7,   KC_P8,   KC_P9,   KC_PAST, KC_PMNS, KC_PPLS, KC_DEL,  \
		_______, KC_PGUP, KC_HOME, KC_UP,   KC_END,  _______, _______, KC_P4,   KC_P5,   KC_P6,   KC_PSLS, KC_PSCR, KC_SCRL, KC_INS,  \
```

fauxpark commented on keyboards/dztech/dz60rgb/keymaps/mekanist/keymap.c:
```diff
@@ -8,37 +8,53 @@
 		_______,    KC_P4,    KC_P5,    KC_P6,    _______,    _______,    _______,    _______,    KC_P4,    KC_P5,    KC_P6,    _______,              KC_PENT,
 		_______,    KC_P1,    KC_P2,    KC_P3,    _______,    _______,    _______,    _______,    KC_P1,    KC_P2,    KC_P3,    _______,              _______,
 		_______,    KC_P0,    KC_PDOT,                        KC_ENT,                             KC_P0,    KC_PDOT,  _______,  _______,              _______),
+<<<<<<< HEAD
+	[_LAYER4] = LAYOUT_60_ansi_arrow( /* WIN */
+		KC_GESC,  KC_1,     KC_2,     KC_3,  KC_4,  KC_5,  KC_6,    KC_7,  KC_8,    KC_9,     KC_0,            KC_MINS,  KC_EQL,  KC_BSPC,
+		KC_TAB,   KC_Q,     KC_W,     KC_E,  KC_R,  KC_T,  KC_Y,    KC_U,  KC_I,    KC_O,     KC_P,            KC_LBRC,  KC_RBRC, KC_BSLASH,
+=======
 	[_LAYER4] = LAYOUT( /* WIN */
 		QK_GESC,  KC_1,     KC_2,     KC_3,  KC_4,  KC_5,  KC_6,    KC_7,  KC_8,    KC_9,     KC_0,            KC_MINS,  KC_EQL,  KC_BSPC,
 		KC_TAB,   KC_Q,     KC_W,     KC_E,  KC_R,  KC_T,  KC_Y,    KC_U,  KC_I,    KC_O,     KC_P,            KC_LBRC,  KC_RBRC, KC_BSLS,
+>>>>>>> upstream/develop
```
```suggestion
	[_LAYER4] = LAYOUT_60_ansi_arrow( /* WIN */
		QK_GESC,  KC_1,     KC_2,     KC_3,  KC_4,  KC_5,  KC_6,    KC_7,  KC_8,    KC_9,     KC_0,            KC_MINS,  KC_EQL,  KC_BSPC,
		KC_TAB,   KC_Q,     KC_W,     KC_E,  KC_R,  KC_T,  KC_Y,    KC_U,  KC_I,    KC_O,     KC_P,            KC_LBRC,  KC_RBRC, KC_BSLS,
```

fauxpark commented on keyboards/dztech/dz60rgb/keymaps/moults31/keymap.c:
```diff
@@ -18,21 +18,26 @@
 #include QMK_KEYBOARD_H
 
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+<<<<<<< HEAD
+    [0] = LAYOUT_60_ansi_arrow(
+        KC_GESC,        KC_1,    KC_2,            KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS,  KC_EQL,  KC_BSPC,
+=======
     [0] = LAYOUT(
         QK_GESC,        KC_1,    KC_2,            KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS,  KC_EQL,  KC_BSPC,
+>>>>>>> upstream/develop
```
```suggestion
    [0] = LAYOUT_60_ansi_arrow(
        QK_GESC,        KC_1,    KC_2,            KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS,  KC_EQL,  KC_BSPC,
```

fauxpark commented on keyboards/dztech/dz60rgb/keymaps/perseid/keymap.c:
```diff
@@ -12,13 +12,18 @@ enum custom_keycodes {
 #define FNM MO(_FNM)
 
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+<<<<<<< HEAD
+	[_QWERTY] = LAYOUT_60_ansi_arrow( /* Base */
+		KC_GESC,  KC_1,     KC_2,     KC_3,  KC_4,  KC_5,  KC_6,    KC_7,  KC_8,    KC_9,     KC_0,       KC_MINS,  KC_EQL,  KC_BSPC,
+=======
 	[_QWERTY] = LAYOUT( /* Base */
 		QK_GESC,  KC_1,     KC_2,     KC_3,  KC_4,  KC_5,  KC_6,    KC_7,  KC_8,    KC_9,     KC_0,       KC_MINS,  KC_EQL,  KC_BSPC,
+>>>>>>> upstream/develop
```
```suggestion
	[_QWERTY] = LAYOUT_60_ansi_arrow( /* Base */
		QK_GESC,  KC_1,     KC_2,     KC_3,  KC_4,  KC_5,  KC_6,    KC_7,  KC_8,    KC_9,     KC_0,       KC_MINS,  KC_EQL,  KC_BSPC,
```

fauxpark commented on keyboards/dztech/dz60rgb/keymaps/piv3rt/keymap.c:
```diff
@@ -54,36 +54,46 @@ enum piv3rt_rgbprofiles {
 enum piv3rt_rgbprofiles current_profile = OFF;
 
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+<<<<<<< HEAD
+    [_DEF] = LAYOUT_60_ansi_arrow(
+        KC_GESC, KC_1,     KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,     KC_MINS,  KC_EQL,  KC_BSPC,
+=======
     [_DEF] = LAYOUT(
         QK_GESC, KC_1,     KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,     KC_MINS,  KC_EQL,  KC_BSPC,
+>>>>>>> upstream/develop
```
```suggestion
    [_DEF] = LAYOUT_60_ansi_arrow(
        QK_GESC, KC_1,     KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,     KC_MINS,  KC_EQL,  KC_BSPC,
```

fauxpark commented on keyboards/dztech/dz60rgb/keymaps/piv3rt/keymap.c:
```diff
@@ -54,36 +54,46 @@ enum piv3rt_rgbprofiles {
         _______,           _______, _______, _______, _______, _______, _______, _______, _______, _______,  _______,  RGB_000, _______,
         _______, _______,  _______,                            RGB_RST,                   _______, _______,  KC_PWR,   AMDREC,  AMDREP
     ),
+<<<<<<< HEAD
+    [_NUM] = LAYOUT_60_ansi_arrow(
+        KC_NLCK, _______,  _______, _______, _______, _______, _______, _______, KC_P7,   KC_P8,   KC_P9,    _______,  _______, _______,
+=======
     [_NUM] = LAYOUT(
         KC_NUM,  _______,  _______, _______, _______, _______, _______, _______, KC_P7,   KC_P8,   KC_P9,    _______,  _______, _______,
+>>>>>>> upstream/develop
```
```suggestion
    [_NUM] = LAYOUT_60_ansi_arrow(
        KC_NUM,  _______,  _______, _______, _______, _______, _______, _______, KC_P7,   KC_P8,   KC_P9,    _______,  _______, _______,
```

fauxpark commented on keyboards/dztech/dz60rgb/keymaps/via/keymap.c:
```diff
@@ -1,36 +1,51 @@
 #include QMK_KEYBOARD_H
 
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+<<<<<<< HEAD
+    [0] = LAYOUT_60_ansi_arrow(
+        KC_GESC,        KC_1,    KC_2,            KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS,         KC_EQL,  KC_BSPC,
+=======
     [0] = LAYOUT(
         QK_GESC,        KC_1,    KC_2,            KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS,         KC_EQL,  KC_BSPC,
+>>>>>>> upstream/develop
```
```suggestion
    [0] = LAYOUT_60_ansi_arrow(
        QK_GESC,        KC_1,    KC_2,            KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS,         KC_EQL,  KC_BSPC,
```

fauxpark commented on keyboards/dztech/dz60rgb/keymaps/via/keymap.c:
```diff
@@ -1,36 +1,51 @@
         _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
         _______, _______, _______,                            _______,                   _______, _______, _______, _______, _______
     ),
+<<<<<<< HEAD
+    [3] = LAYOUT_60_ansi_arrow(
+        KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS, KC_PPLS, _______, _______, _______, _______, KC_PSLS, KC_PAST, KC_PMNS, KC_PPLS, _______,
+=======
     [3] = LAYOUT(
         KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS, KC_PPLS, _______, _______, _______, _______, KC_PSLS, KC_PAST, KC_PMNS, KC_PPLS, _______,
+>>>>>>> upstream/develop
```
```suggestion
    [3] = LAYOUT_60_ansi_arrow(
        KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS, KC_PPLS, _______, _______, _______, _______, KC_PSLS, KC_PAST, KC_PMNS, KC_PPLS, _______,
```

fauxpark commented on keyboards/dztech/dz60rgb/keymaps/via/keymap.c:
```diff
@@ -1,36 +1,51 @@
         _______,          KC_P1,   KC_P2,   KC_P3,   _______, _______, _______, _______, KC_P1,   KC_P2,   KC_P3,   _______, _______,
         _______, KC_P0,   KC_PDOT,                            KC_PENT,                   KC_P0,   KC_PDOT, _______, _______, _______
     ),
+<<<<<<< HEAD
+    [4] = LAYOUT_60_ansi_arrow(
+        KC_GESC,        KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS,         KC_EQL,  KC_BSPC,
+=======
     [4] = LAYOUT(
         QK_GESC,        KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS,         KC_EQL,  KC_BSPC,
+>>>>>>> upstream/develop
```
```suggestion
    [4] = LAYOUT_60_ansi_arrow(
        QK_GESC,        KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS,         KC_EQL,  KC_BSPC,
```

fauxpark commented on keyboards/dztech/dz60rgb/keymaps/xunz/keymap.c:
```diff
@@ -4,11 +4,25 @@
 #define _LAYER2 2
 
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+<<<<<<< HEAD
+	[_LAYER0] = LAYOUT_60_ansi_arrow(KC_GESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSPC,
+=======
 	[_LAYER0] = LAYOUT(QK_GESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSPC,
+>>>>>>> upstream/develop
```
```suggestion
	[_LAYER0] = LAYOUT_60_ansi_arrow(QK_GESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSPC,
```

fauxpark commented on keyboards/dztech/dz60rgb/keymaps/xunz/keymap.c:
```diff
@@ -4,11 +4,25 @@
 							KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_ENT,
 							KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, RSFT_T(KC_SLSH), KC_UP, LT(2,KC_DEL),
 							KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, KC_RALT, MO(1), KC_LEFT, KC_DOWN, KC_RGHT),
+<<<<<<< HEAD
+	[_LAYER1] = LAYOUT_60_ansi_arrow(KC_TRNS, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, KC_DEL,
+							 KC_TRNS, KC_TRNS, KC_UP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_INS, KC_TRNS, KC_PSCR, KC_SLCK, KC_PAUS, QK_BOOT,
+							 KC_TRNS, KC_LEFT, KC_DOWN, KC_RGHT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_HOME, KC_PGUP, EEP_RST,
+							 KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_END, KC_PGDN, KC_VOLU, KC_MUTE,
+								KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_CALC, KC_TRNS, KC_MPRV, KC_VOLD, KC_MNXT),
+	[_LAYER2] = LAYOUT_60_ansi_arrow(KC_TRNS, KC_P1, KC_P2, KC_P3, KC_P4, KC_P5, KC_P6, KC_P7, KC_P8, KC_P9, KC_P0, KC_PMNS, KC_PPLS, KC_DEL,
+								KC_TRNS, RGB_TOG, KC_TRNS, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, RGB_MOD, KC_TRNS, KC_PSLS, KC_PAST, QK_BOOT,
+								KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RGB_SPI, RGB_SPD, KC_TRNS, KC_TRNS, EEP_RST,
+=======
```
```suggestion
```

fauxpark commented on keyboards/dztech/dz60rgb/keymaps/xunz/keymap.c:
```diff
@@ -4,11 +4,25 @@
+								KC_TRNS, RGB_TOG, KC_TRNS, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, RGB_MOD, KC_TRNS, KC_PSLS, KC_PAST, QK_BOOT,
+								KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RGB_SPI, RGB_SPD, KC_TRNS, KC_TRNS, EEP_RST,
+=======
 	[_LAYER1] = LAYOUT(KC_TRNS, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, KC_DEL,
```
```suggestion
	[_LAYER1] = LAYOUT_60_ansi_arrow(KC_TRNS, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, KC_DEL,
```

fauxpark commented on keyboards/dztech/dz60rgb/keymaps/xunz/keymap.c:
```diff
@@ -17,6 +31,7 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
 	[_LAYER2] = LAYOUT(KC_TRNS, KC_P1, KC_P2, KC_P3, KC_P4, KC_P5, KC_P6, KC_P7, KC_P8, KC_P9, KC_P0, KC_PMNS, KC_PPLS, KC_DEL,
 								KC_TRNS, RGB_TOG, KC_TRNS, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, RGB_MOD, KC_TRNS, KC_PSLS, KC_PAST, QK_BOOT,
 								KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RGB_SPI, RGB_SPD, KC_TRNS, KC_TRNS, EE_CLR,
+>>>>>>> upstream/develop
```
```suggestion
	[_LAYER2] = LAYOUT_60_ansi_arrow(KC_TRNS, KC_P1, KC_P2, KC_P3, KC_P4, KC_P5, KC_P6, KC_P7, KC_P8, KC_P9, KC_P0, KC_PMNS, KC_PPLS, KC_DEL,
								KC_TRNS, RGB_TOG, KC_TRNS, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, RGB_MOD, KC_TRNS, KC_PSLS, KC_PAST, QK_BOOT,
								KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RGB_SPI, RGB_SPD, KC_TRNS, KC_TRNS, EE_CLR,
```

arcmags commented:
I'll try to explain here what this pull request is all about, what I believe
the issues are, and what I think should be done or what I'm trying to do.

dztech/dz60rgb_ansi
===================

I noticed that a `LAYOUTS` variable for a keyboard I own is not present in it's
*rules.mk* file while the layout is correctly defined in it's *keyboard.h* and
*info.json* files. This is unfortunate; without this variable set in *rules.mk*
it is not readily possible to use corresponding keymaps in *layouts/community*.

The keyboard in question is: *dztech/dz60rgb_ansi*. When `LAYOUTS` is correctly
set in the various *rules.mk* (3 versions) for this keyboard, keymaps are able
to be compiled from *layouts/community* as expected.

When making these changes, several problems are revealed with existing code.
I will try to document them here. Every build error described below happens in master.

existing errors in master
------------------------

The keymap *dztech/dz60rgb_ansi/keymaps/kuru* fails to build because the
variable `g_suspend_state` is used in *keymap.c* without being declared.

The keymap *dztech/dz60rgb_ansi/keymaps/muralisc* fails to build for the same
reason that *kuru* fails.

The keymap *layouts/community/60_ansi/brandonschlack-ansi* fails to build for
any board with 60_ansi layout because of an `#include rgblight_list.h` that
doesn't exist in *users/brandonschlack/rgb_theme.h*

The kaymap *layouts/community/60_ansi/mechmerlin-ansi* fails to build for any
board with 60_ansi layout because `#define TAB_CODE_DELAY` appears in
*users/mechmerlin/mechmerlin.c* after it's already been defined in
*quantum/action.h*

The keyamp *layouts/community/talljoe-ansi* fails to build for any board with
60_ansi layout with numerous errors relating to to *quantum/keymap_introspection.c*

Because of the above preexisting errors, my pull request is failing checks
unrelated to what I am actually trying to fix.

dztech/dz60rgb
==============

In addition to the *dztech/dz60rgb_ansi* keyboard, I also noticed that no
layout had been specified in any of the files for the *dztech/dz60rgb* keyboard
as well. Fixing this requires changes in more than just *rules.mk* files: the
layout needs to be properly defined in *dz60rgb.h* and every *keymap.c* in this
keyboard folder as well.

When making these changes, more problems were revealed with existing code.

existing errors in master
-------------------------

The keymap *dztech/dz60rgb/keymaps/matthewrobo* fails to build because the
variable `g_suspend_state` is used in *keymap.c* without being declared.

The keymap *dztech/dz60rgb/keymaps/mechmaster48* fails to build because there
is an `/*` within a comment in *keymap.c*. (I corrected this.)

The keymap *dztech/dz60rgb/keymaps/mekanist* fails to build because the
variable `g_suspend_state` is used in *keymap.c* without being declared.

The keymap *dztech/dz60rgb/keymaps/moults31* fails to build because a typo:
`KC_RSHFT` instead of `KC_RSFT`. (I corrected this.)

The keymap *dztech/dz60rgb/keymaps/perseid* fails to build because the
variable `g_suspend_state` is used in *keymap.c* without being declared.

The keymap *dztech/dz60rgb/keymaps/piv3rt* fails to build because there is
`#define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_ALL` in the keyboard
*config.h*, but then in *dztech/dz60rgb/keymaps/piv3rt/config.h* there is
`#undef ENABLE_RGB_MATRIX_CYCLE_ALL`.  `#define RGB_MATRIX_STARTUP_MODE
RGB_MATRIX_CYCLE_ALL` probably shouldn't be in the keyboard *config.h* for this
reason (in case a user undefs it later).  Rather `RGB_MATRIX_STARTUP_MODE`
should either be defined by user keymaps, or allowed to take the fallback def
in *quantum/rgb_matrix/rgb_matrix.c*. (I could correct this, but I am not
certain on how to proceed.)

Merge Conflicts
===============

It looks like some keycodes changed while I've been working on this. I tried to
clear up these merge conflicts as best I could for now.

How to Proceed
==============

I'm not entirely sure, that's why I've taken the time to test all this stuff
out and write it down.  All I was really setting out to do was add the proper
layout to my keyboard's *keyboard.mk* file. It revealed other issues that need
to be addressed, but perhaps those should be in separate pull requests.

Anyway, every error that now occurs in the QM CI Build check with my latest
commit [aa12a33] also occurs in master as described above.

arcmags commented:
I have reduced the scope of my pull request to only three *rules.mk* files.  The errors currently occurring with the QMK CI Build check are all preexisting errors as described above.
